mava/components/building/best_checkpointer.py

```diff
@@ -0,0 +1,74 @@
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Tuple, Union
+
+from optax import Params
+
+from mava.components.component import Component
+from mava.core_jax import SystemBuilder, SystemParameterServer
+
+
+@dataclass
+class BestCheckpointerConfig:
+    checkpointing_metric: Tuple = ("mean_episode_return",)
+    checkpoint_best_perf: bool = False
+
+
+class BestCheckpointer(Component):
+    def __init__(
+        self, config: BestCheckpointerConfig = BestCheckpointerConfig()
+    ) -> None:
+        """Initialise BestCheckpointer"""
+        super().__init__(config)
+
+    def on_building_init(self, builder: SystemBuilder) -> None:
+        """Store checkpointing params in the store"""
+        # Save list of metrics attached with their best performance
+        builder.store.checkpointing_metric: Dict[str, Any] = {}  # type: ignore
+        for metric in list(self.config.checkpointing_metric):
+            builder.store.checkpointing_metric[metric] = None
+
+    def on_building_executor_start(self, builder: SystemBuilder) -> None:
+        """Initialises the store for best model checkpointing"""
+        if not (builder.store.is_evaluator and self.config.checkpoint_best_perf):
+            return
+
+        builder.store.best_checkpoint = self.init_checkpointing_params(builder)
+
+    def on_parameter_server_init(self, server: SystemParameterServer) -> None:
+        """Adding checkpointing parameters to parameter server"""
+        if not self.config.checkpoint_best_perf:
+            return
+
+        server.store.parameters.update(self.init_checkpointing_params(server))
```

Comment:
We need to make separate store variable for the best checkpoint else it will update the same vars related to the current training network